README.md

```diff
@@ -1 +1,89 @@
-# cnp-module-automation-runbook-app-secret-alert
\ No newline at end of file
+## App Registration Alert Module
+
+This Module will create a PowerShell runbook which checks for App Registration Client Secrets that are expiring within 21 days.
+
+```terraform
+module "app_registration_alert" {
+  source = "../"
+
+  automation_account_name = "automation-account"
+  resource_group_name     = "resource-group"
+  location                = "uk south"
+
+  azure_credentials = {
+    description = "azure spn with azure ad access"
+    name        = "service-principal"
+    password    = azuread_application_password.password.value
+    username    = azuread_appplication.app.application_id
+  }
+
+  dynatrace_credentials = {
+    description = "dynatrace api token"
+    name        = "dynatrace-token"
+    password    = azurerm_key_vault_secret.dynatrace_token.value
+    username    = "dynatrace"
+  }
+
+  runbook_parameters = {
+    applicationids      = "000000, 111111"
+    azuretenant         = "000000-0000000-000000"
+    azurecredential     = "service-prinicpal"
```

Comment:
could you make as part of the `azure_credentials` the name mandatory, but the other properties optional.
Therefore, if all details passed then will create a new credential, but if only name then will assume an already create one?
This could save on duplication and confusion.

Same for the dynatrace_credentials

Reply:
I don't like that idea because it makes it confusing.

The way it is done now is clear, there is one variable for creating a credential if required, then another for passing parameters to the runbook.

If you start passing parameters to the runbook from multiple variables makes it more confusing to the user.

Reply:
I have made it so

if `azure_credentials` is present, then the runbook will take the name from that credential block.

if `azure_credentials` is NOT present, it will use var.runbook_parameters.azurecredential